src/storable.rs

```diff
@@ -297,6 +298,39 @@ impl<const N: usize> BoundedStorable for [u8; N] {
     const IS_FIXED_SIZE: bool = true;
 }
 
+impl Storable for Principal {
+    fn to_bytes(&self) -> Cow<[u8]> {
+        Cow::Owned(self.as_slice().to_vec())
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        Self::from_slice(bytes.as_ref())
+    }
+}
+
+impl BoundedStorable for Principal {
+    const MAX_SIZE: u32 = 30;
```

Comment:
Isn't the max size of a principal 29 bytes as per the [interface spec](https://internetcomputer.org/docs/current/references/ic-interface-spec#principal)?

Reply:
https://github.com/dfinity/candid/blob/0ae3c22c6df17f7e09fb49c5ce5fe77653884e10/rust/candid/src/types/principal.rs#L85

Reply:
But you're right. `as_slice` doesn't include the length.